src/sql/workbench/contrib/notebook/browser/calloutDialog/imageCalloutDialog.ts

```diff
@@ -0,0 +1,238 @@
+	private _imageLocationLabel: HTMLElement;
+	private _imageLocalRadioButton: RadioButton;
+	private _editorImageLocationGroup: string = 'editorImageLocationGroup';
+	private _imageRemoteRadioButton: RadioButton;
+	private _imageUrlLabel: HTMLElement;
+	private _imageUrlInputBox: InputBox;
+	private _imageBrowseButton: HTMLAnchorElement;
+	private _imageEmbedLabel: HTMLElement;
+	private _imageEmbedCheckbox: Checkbox;
+	private readonly insertButtonText = localize('callout.insertButton', "Insert");
+	private readonly cancelButtonText = localize('callout.cancelButton', "Cancel");
+	private readonly locationLabel = localize('callout.locationLabel', "Image location");
+	private readonly localImageLabel = localize('callout.localImageLabel', "This computer");
+	private readonly remoteImageLabel = localize('callout.remoteImageLabel', "Online");
+	private readonly pathInputLabel = localize('callout.pathInputLabel', "Image URL");
+	private readonly pathPlaceholder = localize('callout.pathPlaceholder', "Enter image path");
+	private readonly urlPlaceholder = localize('callout.urlPlaceholder', "Enter image URL");
+	private readonly browseAltText = localize('callout.browseAltText', "Browse");
+	private readonly embedImageLabel = localize('callout.embedImageLabel', "Attach image to notebook");
+
+	constructor(
+		title: string,
+		width: DialogWidth,
+		dialogProperties: IDialogProperties,
+		@IPathService private readonly _pathService: IPathService,
+		@IFileDialogService private readonly _fileDialogService: IFileDialogService,
+		@IContextViewService private _contextViewService: IContextViewService,
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IClipboardService clipboardService: IClipboardService,
+		@ILogService logService: ILogService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService
+	) {
+		super(
+			title,
+			width,
+			dialogProperties,
+			themeService,
+			layoutService,
+			telemetryService,
+			contextKeyService,
+			clipboardService,
+			logService,
+			textResourcePropertiesService
+		);
+
+		this._selectionComplete = new Deferred<ICalloutDialogOptions>();
+	}
+
+	/**
+	 * Opens the dialog and returns a promise for what options the user chooses.
+	 */
+	public open(): Promise<ICalloutDialogOptions> {
+		this.show();
+		return this._selectionComplete.promise;
+	}
+
+	public render() {
+		super.render();
+		attachModalDialogStyler(this, this._themeService);
+		this.addFooterButton(this.insertButtonText, () => this.insert());
+		this.addFooterButton(this.cancelButtonText, () => this.cancel(), undefined, true);
+		this.registerListeners();
+	}
+
+	protected renderBody(container: HTMLElement) {
+		this.buildInsertImageCallout(container);
+	}
+
+	private buildInsertImageCallout(container: HTMLElement): void {
```

Comment:
This can just be in the renderBody function - no real point in having a separate function now that it's in its own class